packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -217,7 +214,13 @@ - (void)restoreTransactions:(FlutterMethodCall *)call result:(FlutterResult)resu
               details:call.arguments]);
     return;
   }
-  [self.paymentQueueHandler restoreTransactions:call.arguments];
+  @try {
+    [self.paymentQueueHandler restoreTransactions:call.arguments];
+  } @catch (NSException *e) {
```

Comment:
Yes, there are more types of exceptions. I can check what was thrown when network is not available. However, I feel like we should still catch all the exceptions here, and send it to dart so our users won't crash. What do you think?

Reply:
In general I agree, but here I'm wary of preventing a crash if we're not sure of what the reason for the crash is. Don't want to hide something that actually legitimately could be crash-worthy here. We know that we don't want to throw for network problems, but I'm not sure what else could be causing an issue here and don't want to silently pave over anything that's a real problem.

Reply:
I realized that this is not necessary. Since the error of restoretranscations is passed to the delegate method, and we have already been passing the error from objc to dart there.
So we only need to add this error to the response object that we newly created in from dart side.